### DISCUSSION TITLE
Build Systems

### DISCUSSION THREAD

**Discussion by zachgrayio** [MobileNativeFoundation/discussions#41] [General]
I'm hesitant to post this, but heading off the [inevitable](https://github.com/MobileNativeFoundation/discussions/discussions/6#discussioncomment-427931) 😅 

Lest this thread descend directly into madness, let's establish some ground-rules perhaps? 

Maybe something like simple information gathering to start with? Something along the lines of

- optional: company/companies, if authorized to share
- size of codebases and/or teams
- host platforms built on
- target platforms built for
- third-party build systems in use (or, if standard tooling has held up at scale)
- third-party build systems evaluated, rejected, or _ejected_, and why
- advanced **build** tooling in place (shared cache, remote build exec, metrics and events gathering, build graph analysis)
- typical local and CI build and test times (avoid covering ground already touched in other threads)

(let's amend this list as needed, just let me know in comments before we kick off posting responses)

Disclaimer: I'm [very bullish on Bazel](https://bazel.build/experts.html) but [we try not to](https://flare.build/blog/flare-more-than-a-bazel-company) play favorites here.

**Comment by piannaf** [comment]
This topic is nearly a year old, but I think it's the right place to post this

https://www.reddit.com/r/RedditEng/comments/syz5dw/ios_and_bazel_at_reddit_a_journey/

Hits on most if not all the points above

**Reply by mattrobmattrob** [reply]
Let me know on the post or here if there are questions as well. 😄

**Reply by zachgrayio** [reply]
great post!

**Reply by tinder-maxwellelliott** [reply]
> Let me know on the post or here if there are questions as well. 😄

Thanks for the bazel-diff shout-out 😊 !!! Loved the article

**Reply by scottluxenberg** [reply]
This was interesting! @mattrobmattrob will there ever be plans to opensource XcodeGenGen?

**Reply by mattrobmattrob** [reply]
> This was interesting! @mattrobmattrob will there ever be plans to opensource XcodeGenGen?

Working on getting it ready now. Stay tuned.

**Comment by lawgimenez** [comment]
Awesome post! I was wondering how did you get the **code line count**? What tools if any. Thanks!

**Reply by tinder-maxwellelliott** [reply]
We use this http://cloc.sourceforge.net/. Seems to work well for us

**Reply by mattrobmattrob** [reply]
We also use cloc. It's easy to use the `--json` argument to pass the output to `jq` to pull out the things you care about.
